src/js/base/Pagination.jsx

```diff
@@ -36,33 +45,62 @@ const usePagination = (pageCount, currentPage) => {
     return paginationRange;
 };
 
-export const Pagination = ({ currentPage, pageCount, onPageChange }) => {
+export const Pagination = ({ documents, renderRow, currentPage, pageCount, onLoadNextPage }) => {
     const paginationRange = usePagination(pageCount, currentPage);
+    const location = useLocation();
+    const page = new URLSearchParams(location.search).get("page");
+
+    const history = useHistory();
+    useEffect(() => {
+        if (page > 1 && page > pageCount) {
+            history.push({ search: `?page=${page - 1}` });
+        } else {
+            history.push({ search: `?page=${page}` });
+        }
+    }, [pageCount]);
```

Comment:
I think you could simplify this by using the min function. Something like:
```suggestion
    useEffect(() => {
        if (page > pageCount) {
            history.push({ search: `?page=${min(page, pageCount}` });
        } 
    }, [pageCount]);
```